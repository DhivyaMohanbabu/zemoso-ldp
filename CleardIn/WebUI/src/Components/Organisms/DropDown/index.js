/* eslint-disable no-unused-expressions */
/* eslint-disable react/prop-types */
/* eslint-disable max-len */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import { makeStyles } from '@material-ui/core/styles';
import MenuItem from '@material-ui/core/MenuItem';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import { InputLabel } from '@material-ui/core';
import SearchBar from '../../Molecules/SearchBar';
import IconicOption from '../../Molecules/IconicOption';
import NewEntry from '../../Molecules/NewEntry';
import ColoredOption from '../../Atoms/ColoredOption';

const useStyles = makeStyles((theme) => ({
  root: {
    minWidth: 120,
    maxWidth: 150,
  },
  options: {
    display: 'flex',
    flexDirection: 'column',
    marginBottom: 4,

  },
  chipred: {
    margin: 2,
    backgroundColor: theme.palette.primary.main,
  },
  chip: {
    margin: 2,
    backgroundColor: (props) => props.color,
  },
}));

// eslint-disable-next-line react/prop-types
const DropDown = ({
  names, isSecurityDropDown, value, handleGroupChange,
  data, labelFlag,
}) => {
  const classes = useStyles(names);
  const [option, setOption] = useState('');
  const [input, setInput] = useState('');
  if (option === '') {
    names.map((name) => (
      name.label.includes(value) && setOption(name)
    ));
  }

  const handleChange = (event) => {
    setOption(event.target.value);
    handleGroupChange(event.target.value, data);
  };
  const onChange = (event) => {
    setInput(event.target.value);
  };
  const icon = isSecurityDropDown ? (ArrowDropDownIcon) : (MoreVertIcon);
  const flag = true;
  return (
    <div>
      <FormControl className={classes.root}>
        {labelFlag && (<InputLabel id="select-label">TrustGroup</InputLabel>)}
        <Select
          // eslint-disable-next-line react/jsx-boolean-value
          autoWidth={true}
          data-testid="select"
          disableUnderline={flag}
          labelId="select-label"
          id="demo-controlled-open-select"
          value={option}
          onChange={handleChange}
          IconComponent={icon}
          renderValue={(selected) => (
            isSecurityDropDown
              ? (<ColoredOption key={selected.label} label={selected.label} className={classes.chip} />) : (<IconicOption icon={selected.icon} label={selected.label} />)
          )}

        >
          {isSecurityDropDown ? (
            <SearchBar placeholder="find option" onChange={onChange} />) : (<SearchBar placeholder="Enable  Department Access" onChange={onChange} />)}
          {names.map((name) => (

            name.label.includes(input) && (
            <MenuItem key={name.label} value={name}>
              {isSecurityDropDown ? (
                <ColoredOption key={name.label} label={name.label} />)
                : (<IconicOption icon={name.icon} label={name.label} />)}
            </MenuItem>
            )
          ))}
          {!isSecurityDropDown && (
          <MenuItem>
            <NewEntry text="New Department" />
          </MenuItem>
          )}
        </Select>
      </FormControl>

    </div>
  );
};
DropDown.prototype = {
  names: PropTypes.arrayOf(Object).isRequired,
};
export default DropDown;
