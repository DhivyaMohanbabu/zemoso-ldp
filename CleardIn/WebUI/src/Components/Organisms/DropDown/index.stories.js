import { orange } from '@material-ui/core/colors';
import React from 'react';
import ViewStreamOutlinedIcon from '@material-ui/icons/ViewStreamOutlined';
import DropDown from '.';

export default {
  title: 'Organism/DropDown',
};

const Template = (args) => <DropDown {...args} />;

export const SecurityDropDown = Template.bind({});
SecurityDropDown.args = {
  names: [{ label: 'Top 100', color: 'red' }, { label: 'Top 200', color: orange[500] }],
  isSecurityDropDown: true,
  value: 'Top 100',
  labelFlag: false,
};
export const DepartmentDropDown = Template.bind({});
DepartmentDropDown.args = {
  names: [{ label: 'Sales', icon: ViewStreamOutlinedIcon }, { label: 'Department', icon: ViewStreamOutlinedIcon }],
  isSecurityDropDown: false,
};
