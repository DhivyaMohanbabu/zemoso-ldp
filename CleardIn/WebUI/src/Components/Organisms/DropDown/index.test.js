/* eslint-disable max-len */
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import ViewStreamOutlinedIcon from '@material-ui/icons/ViewStreamOutlined';
import DropDown from '.';
import '@testing-library/jest-dom';

describe('security OPtion', () => {
  it('should render new Row ', () => {
    const names = [{ label: 'Top100', color: 'red' }, { label: 'Top 200', color: 'blue' }];
    const flag = true;
    const container = render(<DropDown names={names} isSecurityDropDown={flag} handleGroupChange={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });

  it('should render Security Option ', () => {
    const names = [{ label: 'Top100', color: 'red' }];
    const flag = true;
    const { queryByTestId } = render(<DropDown
      names={names}
      isSecurityDropDown={flag}
      handleGroupChange={jest.fn()}
    />);
    fireEvent.change(screen.getByTestId('select'), {
      target: { label: 'Top100', color: 'red' },
    });
    const mySelectComponent = queryByTestId('select');
    const selectNode = mySelectComponent.childNodes[0].childNodes[0];
    fireEvent.change(selectNode, { target: { label: 'Top100', color: 'red' } });
    fireEvent.click(mySelectComponent);
    expect(mySelectComponent).toBeDefined();
    expect(mySelectComponent).not.toBeNull();
  });
  it('should render department Option ', () => {
    const names = [{ label: 'Top100', icon: ViewStreamOutlinedIcon }];
    const flag = false;
    const { getByText, queryByTestId } = render(<DropDown
      names={names}
      isSecurityDropDown={flag}
      handleGroupChange={jest.fn()}
    />);
    fireEvent.change(screen.getByTestId('select'), {
      target: { label: 'Top100', icon: ViewStreamOutlinedIcon },
    });
    const mySelectComponent = queryByTestId('select');
    const selectNode = mySelectComponent.childNodes[0].childNodes[0];
    fireEvent.change(selectNode, { target: { label: 'Top100', icon: ViewStreamOutlinedIcon } });
    expect(mySelectComponent).toBeDefined();
    expect(mySelectComponent).not.toBeNull();
    fireEvent.mouseEnter(queryByTestId('select'));
    fireEvent.keyDown(mySelectComponent.firstChild, { key: 'ArrowDown' });
    fireEvent.click(getByText('Top100'));
  });

  it('should render Security Option ', () => {
    const mock = jest.fn();
    expect(mock.mock.calls).toEqual([]);
  });
});
