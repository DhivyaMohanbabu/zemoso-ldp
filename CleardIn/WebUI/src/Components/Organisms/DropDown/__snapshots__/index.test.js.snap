// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`security OPtion should render new Row  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiFormControl-root makeStyles-root-1"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
            data-testid="select"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="select-label demo-controlled-open-select"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="demo-controlled-open-select"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiFormControl-root makeStyles-root-1"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl"
          data-testid="select"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="select-label demo-controlled-open-select"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="demo-controlled-open-select"
            role="button"
            tabindex="0"
          >
            <span>
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
